File: 1-TwoSum/1-TwoSum.js
// Last updated: 29/05/2025, 21:43:05
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let result = []
    let cache = new Map()
    nums.forEach((num, i) => {
        if (result.length === 2) return
        let cachedNumber = cache.get(target - num)
        if (cachedNumber !== undefined && cachedNumber !== i) {
            return result.push(i, cachedNumber)
        }
        cache.set(num, i)
    })
    return result
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 29/05/2025, 21:43:04
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}
        for idx, num in enumerate(nums):
            complement = target - num
            if complement in cache:
                return [cache[complement], idx]
            cache[num] = idx
        return []
        
        